erpnext/controllers/accounts_controller.py

```diff
@@ -1309,45 +1309,28 @@ def add_taxes_from_tax_template(child_item, parent_doc):
 					})
 				tax_row.db_insert()
 
-def set_sales_order_defaults(parent_doctype, parent_doctype_name, child_docname, trans_item):
+def set_order_defaults(parent_doctype, parent_doctype_name, child_doctype, child_docname, trans_item):
 	"""
-	Returns a Sales Order Item child item containing the default values
+	Returns a Sales/Purchase Order Item child item containing the default values
 	"""
 	p_doc = frappe.get_doc(parent_doctype, parent_doctype_name)
-	child_item = frappe.new_doc('Sales Order Item', p_doc, child_docname)
+	child_item = frappe.new_doc(child_doctype, p_doc, child_docname)
 	item = frappe.get_doc("Item", trans_item.get('item_code'))
-	child_item.item_code = item.item_code
-	child_item.item_name = item.item_name
-	child_item.description = item.description
-	child_item.delivery_date = trans_item.get('delivery_date') or p_doc.delivery_date
+	for field in ("item_code", "item_name", "description", "item_group"):
+	    child_item.update({field: item.get(field)})
+	date_fieldname = "delivery_date" if child_doctype == "Sales Order Item" else "schedule_date"
+	child_item.date_fieldname = trans_item.get(date_fieldname) or p_doc.date_fieldname
```

Comment:
Cant access attributes that do exist by dot notation. Also you cannot use dot notation to chain a string value. travis is  breaking due to this.
`date_fieldname` does not get substituted here.
```suggestion
	child_item.update({date_fieldname: trans_item.get(date_fieldname) or p_doc.date_fieldname})
```